docs/source-pytorch/common/trainer.rst

```diff
@@ -1231,27 +1231,26 @@ Properties
 callback_metrics
 ****************
 
-The metrics available to callbacks. These are automatically set when you log via `self.log`
+.. autoproperty:: lightning.pytorch.trainer.Trainer.callback_metrics
+   :noindex:
 
-.. code-block:: python
+logged_metrics
+**************
 
-    def training_step(self, batch, batch_idx):
-        self.log("a_val", 2)
+.. autoproperty:: lightning.pytorch.trainer.Trainer.logged_metrics
+   :noindex:
 
+progress_bar_metrics
+********************
 
-    callback_metrics = trainer.callback_metrics
-    assert callback_metrics["a_val"] == 2
+.. autoproperty:: lightning.pytorch.trainer.Trainer.progress_bar_metrics
+   :noindex:
 
 current_epoch
 *************
 
-The number of epochs run.
-
-.. code-block:: python
-
-    if trainer.current_epoch >= 10:
-        ...
-
+.. autoproperty:: lightning.pytorch.trainer.Trainer.current_epoch
+   :noindex:
```

Comment:
Note that this will put all the ugly formatting here, including types etc. 
It is one of the reasons why we didn't just autogenerate this section. It is more digestible when you just document the essential usage here.

For full reference, the curious user can always go to the full api reference and code.

Reply:
<img width="708" alt="image" src="https://user-images.githubusercontent.com/5495193/223434887-56ec82c7-f4ce-47de-a8f9-9c70e5095fe4.png">

Reply:
What do you suggest then? That we duplicate the information?

Reply:
Yes, why not :) Didn't you hear, after 2.0, Trainer is anyway done. No API changes allowed 🤣 

I think in general, if it minimizes maintenance effort, but makes the user experience worse, we should not do it.
Alternatively, we could invest some time in making a custom template directive for rendering this output, if we believe it is important.

Reply:
I spent a while trying a solution, even asked ChatGPT and couldn't find one.

Do you happen to know where are the directives that render `.. automethod::`? Those look fine, so it should be similar for `.. autoproperty::`

Reply:
There are some templates in the folder _templates. For example, there is a classtemplate.rst, which you can see if you search for it, gets added to some directives like this:

```
.. autosummary::
    :nosignatures:
    :template: classtemplate.rst
```
    
You might be able to make a template for automethod following this pattern, not sure though.

Reply:
I failed to find a solution. We would need support from a Sphinx wizard. I copied the contents